pynpoint/processing/psfpreparation.py

```diff
@@ -104,9 +102,19 @@ def run(self) -> None:
 
         # Get the number of images and split into batches to comply with memory constraints
         im_shape = self.m_image_in_port.get_shape()
-        nimages = im_shape[0]
+        nimages = im_shape[-3]
         frames = memory_frames(memory, nimages)
 
+        # Read out number of wavelength
+        if len(im_shape) < 4:
+            wavelength_dim = 1
+        else:
+            wavelength_dim = im_shape[-4]
+
+            # Set output shape
+            tmp_output = np.zeros(im_shape)
+            self.m_image_out_port.set_all(tmp_output, keep_attributes=False)
```

Comment:
Please do not use `set_all` or `get_all` unless strictly necessary. Instead, append the data to the output port in amounts of `MEMORY`.

Reply:
The problem here is that the `memory_frames` only splits along 1 dimension (time) where IFS data has two dimensions (wavelength, time). I could work around this, but it might be cleaner if the `memory_frames` function would be adjusted.

The second problem is that I'm not sure if append works with IFS data. For non-IFS data, append can simply add an arbitrary number of frames along the time dimension. With IFS data, even though you can add along the wavelength dimension, you would need all time frames already prepared to be added. The only work around possible I found was the one I used in lines 164-167.

Reply:
Perhaps we could process and append the data per wavelength. So in case of IFS data, all time steps are processed at once. So it will not use `MEMORY`, but that is perhaps also less critical with NIR data.

Reply:
That is defentily a good option. I will implement it like this for now, but I'll keep thinking about it.

Reply:
I tried the implementation but using `self.m_image_out_port.append(images, data_dim=4)` appends along the second dimension instead of the first. Could you look into this?

Reply:
Right, that is indeed the case in `dataio`. Maybe that is required by the PCA module, but I am not sure. We could add a parameter, but perhaps the easiest solution would be to process al the wavelengths per exposure at once (instead of the other way around, as I previously suggested)? Would that work?

Reply:
You are right! I miss understood you the first time but now implemented it as you mentioned it and it works fine.